### PR TITLE
chore(ci): increase timeout for docker container to come online

### PR DESCRIPTION
**Pull request by JohnVillalovos**
Have been seeing timeout issues more and more. Increase timeout from
200 seconds to 300 seconds (5 minutes).

### CODE REVIEW

**Comment by nejch** [comment]
Thanks @JohnVillalovos!

What I've also been thinking is that sometimes the GitLab container completely fails to start up properly in underpowered CI VMs (I have a similar test setup on GitLab CI using `services:`).

If that still happens with 5 minutes, we might need to investigate further slimming down the services started in the Omnibus config. GitLab is quite a beast now with all the batteries-included stuff :thinking:

File: tests/functional/conftest.py [hunk 1]
```diff
@@ -191,11 +191,16 @@ def gitlab_config(check_is_alive, docker_ip, docker_services, temp_dir, fixture_
...
     logging.info("Waiting for GitLab container to become ready.")
     docker_services.wait_until_responsive(
-        timeout=200, pause=10, check=lambda: check_is_alive("gitlab-test")
+        timeout=300, pause=10, check=lambda: check_is_alive("gitlab-test")
+    )
+    setup_time = time.perf_counter() - start_time
+    logging.info(
+        f"GitLab container is now ready after {setup_time:0.2f} seconds "
+        f"({setup_time/60:0.2f} minutes)."
     )
```
[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply tests/functional/conftest.py:203]: I like minutes!!!

[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply tests/functional/conftest.py]: Sort of done like you asked. But a bit different.

[the remark this answers is not in the document]
Comment (nejch) [reply tests/functional/conftest.py]: :sweat_smile: I was hoping to reduce the code, not add more :grin:

File: tests/functional/conftest.py [hunk 2]
```diff
@@ -154,8 +154,13 @@ def check_is_alive():
...
 
-    def _check(container):
-        logging.info("Checking if GitLab container is up...")
+    def _check(container: str, start_time: float) -> bool:
+        setup_time = time.perf_counter() - start_time
+        minutes, seconds = int(setup_time / 60), int(setup_time % 60)
+        logging.info(
+            f"Checking if GitLab container is up. "
+            f"Have been checking for {minutes} minute(s), {seconds} seconds ..."
+        )
```
Comment (nejch) [inline tests/functional/conftest.py:163]: IMO for these we have the logging timestamps. Our functional test `conftest.py` is getting quite bloated just for the container startup compared to things like https://github.com/gitlabhq/terraform-provider-gitlab/blob/main/scripts/await-healthy.sh.

Comment (JohnVillalovos) [reply tests/functional/conftest.py:163]: True, we do have timestamps. But then it means I need to do math in my head. I hate doing math in my head 😉

**Pull request merged**